spec/reservation_manager_spec.rb

```diff
@@ -0,0 +1,122 @@
+  end
+
+  it "raises ArgumentError when date range is invalid" do
+    id = 1
+    start_time = Time.parse("2019-03-19 14:08:45 -0700")
+    end_time = Time.parse("2019-03-15 14:08:45 -0700")
+    room_number = 3
+
+    expect { Hotel::ReservationManager.new(start_time, end_time, room_number) }.must_raise ArgumentError
+  end
+
+  it "creates a reservation for an available room" do
+    reservation_generator
+    reservation = manager_new.create_reservation(Time.parse("2019-03-21 14:08:45 -0700"), Time.parse("2019-03-26 14:08:45 -0700"), 6)
+    expect(reservation).must_be_instance_of Hotel::Reservation
+  end
+
+  it "raises ArgumentError when a room chosen is not existent" do
+    expect { manager_new.create_reservation(Time.parse("2019-03-21 14:08:45 -0700"), Time.parse("2019-03-26 14:08:45 -0700"), 24) }.must_raise ArgumentError
+  end
+
+  it "raises ArgumentError when a room chosen is not available" do
+    reservation_generator
+
+    expect { manager_new.create_reservation(Time.parse("2019-03-19 14:08:45 -0700"), Time.parse("2019-03-26 14:08:45 -0700"), 6) }.must_raise ArgumentError
+  end
+
+  it "raises ArgumentError when range of dates are invalid" do
+    start_date = Time.parse("2019-03-19 14:08:45 -0700")
+    end_date = Time.parse("2019-03-17 14:08:45 -0700")
+    room = 4
+    expect { manager_new.create_reservation(start_date, end_date, room) }.must_raise ArgumentError
+  end
+
+  it "shows the list of rooms in the hotel" do
+    expect(manager_new.rooms).must_be_kind_of Array
+  end
+
+  it "stores the reservations" do
+    reservation_generator
+
+    expect(manager_new.reservations).must_be_kind_of Array
+    expect(manager_new.reservations.length).must_equal 4
+  end
+
+  it "finds a reservation by id" do
+    reservation_generator
+
+    expect(manager_new.find_by_id(id: 1).start_date).must_equal Time.parse("2019-03-11 14:08:45 -0700")
+    expect(manager_new.find_by_id(id: 2)).must_be_kind_of Hotel::Reservation
+    expect(manager_new.find_by_id(id: 3).start_date).must_equal Time.parse("2019-02-27 14:08:45 -0700")
+    expect(manager_new.find_by_id(id: 4)).must_be_kind_of Hotel::Reservation
+  end
+
+  it "raises ArgumentError when the id is not valid" do
+    reservation_generator
+
+    expect { manager_new.find_by_id(id: 8) }.must_raise ArgumentError
+  end
+
+  it "finds reservations by date" do
+    reservation_generator
+
+    expect(manager_new.find_by_date(Time.parse("2019-03-19 14:08:45 -0700"), Time.parse("2019-03-22 14:08:45 -0700"))).must_be_kind_of Array
+    expect(manager_new.find_by_date(Time.parse("2019-02-27 14:08:45 -0700"), Time.parse("2019-02-28 14:08:45 -0700"))).must_be_kind_of Array
+  end
+
+  it "raises ArgumentError when there are no reservations for date range" do
+    reservation_generator
+
+    expect { manager_new.find_by_date(Time.parse("2019-01-01 14:08:45 -0700"), Time.parse("2019-01-20 14:08:45 -0700")) }.must_raise ArgumentError
```

Comment:
I would suggest that this method should simply return an empty array when no reservation falls in the date range.